src/journal.rs

```diff
@@ -0,0 +1,151 @@
+use std::fs::File;
+use std::fmt;
+use std::io::Write;
+use std::os::unix::io::FromRawFd;
+use std::ffi::CString;
+use std::os::unix::net::UnixDatagram;
+use std::os::unix::io::AsRawFd;
+use nix::sys::socket::{ControlMessage, MsgFlags, sendmsg};
+use nix::sys::memfd::MemFdCreateFlag;
+use nix::sys::memfd::memfd_create;
+use nix::fcntl::*;
+
+use crate::errors::*;
+
+const SD_JOURNAL_SOCK_PATH: &str = "/run/systemd/journal/socket";
+
+/// Log priority values.
+///
+/// See `man 3 syslog`.
+pub enum Priority {
+    /// system is unusable
+    Emergency = 0,
+    /// action must be taken immediately
+    Alert,
+    /// critical conditions
+    Critical,
+    /// error conditions
+    Error,
+    /// warning conditions
+    Warning,
+    /// normal, but significant, condition
+    Notice,
+    /// informational message
+    Info,
+    /// debug-level message
+    Debug,
+}
+
+#[inline(always)]
+fn is_valid_char(c: char) -> bool {
+    if c.is_uppercase() || c.is_numeric() || c == '_' {
```

Comment:
I think you can drop the `if ... {}` and `return`s. Just the condition itself is a valid statement which returns a `bool`.